kDriveCore/Data/UploadQueue/Queue/UploadParallelismHeuristic.swift

```diff
@@ -86,7 +86,15 @@ final class UploadParallelismHeuristic {
         }
 
         // Scaling with the number of activeProcessor
-        currentParallelism = max(4, processInfo.activeProcessorCount)
+        let parallelism = max(4, processInfo.activeProcessorCount)
+
+        // Beginning with .serious state, we start reducing the load on the system
+        guard thermalState != .serious else {
+            currentParallelism = parallelism / 2
```

Comment:
Maybe put a min to be sure to always have a correct value for devices with low core count ?